source/common/http/conn_pool_grid.h

```diff
@@ -206,9 +212,9 @@ class ConnectivityGrid : public ConnectionPool::Instance,
   // that specifies HTTP/3 and HTTP/3 is not broken.
   bool shouldAttemptHttp3();
 
-  // Creates the next pool in the priority list, or nullptr if all pools have been created.
-  // TODO(alyssawilk) replace this now we have explicit pools.
-  virtual ConnectionPool::Instance* createNextPool();
+  // Creates the requested pool
+  virtual ConnectionPool::Instance* getOrCreateHttp3Pool();
+  virtual ConnectionPool::Instance* getOrCreateHttp2Pool();
```

Comment:
OOC, can these methods return null? If not, maybe they should return refs? (But I kinda think they can return null?)

Reply:
they shouldn't return null but the callers want pointers so I didn't think changing the types made sense?